### PR TITLE
Fixing account naming

### LINKED ISSUE #1808: CLI Account New does not work on Windows
latest master, win 7
```
$ ./target/release/parity.exe account new
Please note that password is NOT RECOVERABLE.
Type password:
Repeat password:
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: SStore(Io(Error { repr: Os { code: 123, message: "The filename, directory name, or volume label syntax is incorrect." } }))', ../src/libcore\result.rs:785
stack backtrace:
   0:        0x140639062 - std::rt::lang_start::hbcefdc316c2fbd45
   1:        0x14063864d - std::rt::lang_start::hbcefdc316c2fbd45
   2:        0x1406183cd - std::panicking::rust_panic_with_hook::h5dd7da6bb3d06020
   3:        0x14063b40b - rust_begin_unwind
   4:        0x14061984f - std::panicking::begin_panic_fmt::haf08a9a70a097ee1
   5:        0x14063b04c - rust_begin_unwind
   6:        0x14063f295 - core::panicking::panic_fmt::h93df64e7370b5253
   7:        0x13feb807c - unwrap_failed<ethcore::account_provider::Error>
                        at c:\parity\src\libcore\macros.rs:29
   8:        0x13fcb7598 - execute
                        at c:\parity\src\libcore\result.rs:723
   9:        0x13fd33392 - main
                        at c:\parity\parity\parity\main.rs:113
  10:        0x14063802c - std::rt::lang_start::hbcefdc316c2fbd45
  11:        0x14063e7d1 - _rust_maybe_catch_panic
  12:        0x140637d64 - std::rt::lang_start::hbcefdc316c2fbd45
  13:        0x1407e5777 - __scrt_common_main_seh
                        at f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:264
  14:         0x76e75a4c - BaseThreadInitThunk

```

### LINKED ISSUE #1809: Each run creates new key entries in ~/.parity/keys
```
$ ls ~/.parity/keys/
378fcfa8-f503-2935-75d2-288271817d02                            76926dee-b11e-4dac-9056-8c187977dac9-2016-08-02_10:39:16_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3                            76926dee-b11e-4dac-9056-8c187977dac9-2016-08-02_10:52:47_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-01_14:18:34_.json  76926dee-b11e-4dac-9056-8c187977dac9-2016-08-02_10:52:55_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-01_14:29:28_.json  76926dee-b11e-4dac-9056-8c187977dac9-2016-08-02_11:18:02_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-01_14:32:51_.json  76926dee-b11e-4dac-9056-8c187977dac9-2016-08-02_11:18:10_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-01_14:58:55_.json  76926dee-b11e-4dac-9056-8c187977dac9-2016-08-02_11:18:38_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-01_15:08:19_.json  76926dee-b11e-4dac-9056-8c187977dac9-2016-08-02_11:34:22_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-01_21:23:29_.json  76926dee-b11e-4dac-9056-8c187977dac9-2016-08-02_11:34:33_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-01_21:23:51_.json  76926dee-b11e-4dac-9056-8c187977dac9-2016-08-02_11:35:08_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-01_21:24:03_.json  76926dee-b11e-4dac-9056-8c187977dac9-2016-08-02_11:42:43_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-02_09:40:58_.json  76926dee-b11e-4dac-9056-8c187977dac9-2016-08-02_11:43:02_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-02_10:38:51_.json  76926dee-b11e-4dac-9056-8c187977dac9-2016-08-02_11:45:21_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-02_10:39:16_.json  8bfb9d13-362b-e4e0-25e2-541fd197ed66
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-02_10:52:47_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-02_10:52:55_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-01_14:18:34_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-02_11:18:02_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-01_14:29:28_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-02_11:18:10_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-01_14:32:51_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-02_11:18:38_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-01_14:58:55_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-02_11:34:22_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-01_15:08:19_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-02_11:34:33_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-01_21:23:29_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-02_11:35:08_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-01_21:23:51_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-02_11:42:43_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-01_21:24:03_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-02_11:43:02_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-02_09:40:58_.json
4d89372d-9af7-4d64-9665-9e547f6e5ea3-2016-08-02_11:45:21_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-02_10:38:51_.json
6bab5e93-bf78-3ac0-45cf-de9460ef9754                            e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-02_10:39:16_.json
76926dee-b11e-4dac-9056-8c187977dac9                            e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-02_10:52:47_.json
76926dee-b11e-4dac-9056-8c187977dac9-2016-08-01_14:18:34_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-02_10:52:55_.json
76926dee-b11e-4dac-9056-8c187977dac9-2016-08-01_14:29:28_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-02_11:18:02_.json
76926dee-b11e-4dac-9056-8c187977dac9-2016-08-01_14:32:51_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-02_11:18:10_.json
76926dee-b11e-4dac-9056-8c187977dac9-2016-08-01_14:58:55_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-02_11:18:38_.json
76926dee-b11e-4dac-9056-8c187977dac9-2016-08-01_15:08:19_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-02_11:34:22_.json
76926dee-b11e-4dac-9056-8c187977dac9-2016-08-01_21:23:29_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-02_11:34:33_.json
76926dee-b11e-4dac-9056-8c187977dac9-2016-08-01_21:23:51_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-02_11:35:08_.json
76926dee-b11e-4dac-9056-8c187977dac9-2016-08-01_21:24:03_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-02_11:42:43_.json
76926dee-b11e-4dac-9056-8c187977dac9-2016-08-02_09:40:58_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-02_11:43:02_.json
76926dee-b11e-4dac-9056-8c187977dac9-2016-08-02_10:38:51_.json  e2ee45ff-a1dc-4b77-9aee-c6a84ee460f3-2016-08-02_11:45:21_.json
```

### PR DESCRIPTION
**Pull request by tomusdrw**
1. `:` is not valid part of filename on windows
2. `From<KeyFile>` implementation removed and replaced with `SafeAccount::from_file` usage.

I'm still doing some more tests to confirm that those issues are solved:
Closes #1808 
Closes #1809

### CODE REVIEW

**Comment by gavofyork** [comment]
would be good to more or lesss copy geth's format and document our decision on the wiki:

`UTC--2016-05-13T15-03-10.137084514Z--3e...41`

that said, we might leave out the nanoseconds.

**Comment by tomusdrw** [comment]
Ok, should we have a migration for old keys? Or is it only for newly created?

**Comment by gavofyork** [comment]
only for newly created - should only require alteration of the `format` line.

**Comment by tomusdrw** [comment]
Done. It required alterations of geth import logic as well (so that we don't duplicate account files).

File: ethstore/src/dir/disk.rs [hunk 1]
```diff
@@ -98,22 +100,26 @@ impl KeyDirectory for DiskDirectory {
...
 
 		// build file path
-		let mut account = account;
-		account.path = account.path.or_else(|| {
-			let mut keyfile_path = self.path.clone();
-			let timestamp = time::strftime("%Y-%m-%d_%H:%M:%S_%Z", &time::now()).unwrap_or("???".to_owned());
-			keyfile_path.push(format!("{}-{}.json", keyfile.id, timestamp));
-			Some(keyfile_path)
+		let filename = account.filename.as_ref().cloned().unwrap_or_else(|| {
+			let timestamp = time::strftime("%Y-%m-%dT%H-%M-%S", &time::now_utc()).unwrap_or("???".to_owned());
```
Comment (gavofyork) [inline ethstore/src/dir/disk.rs]: are `?`s a valid component of file names on all platforms?

**Pull request merged**